### ISSUE TITLE
Tailoring kit mods don't improve armor

### ISSUE THREAD

**Issue by l29ah** [CleverRaven/Cataclysm-DDA#48781]
# Describe the bug

# Steps To Reproduce
Load the game.
`a`ctivate the tailoring kit.
Choose `Modify Clothing`.
Select `++ rain hood`.
Choose `pad with Kevlar`.
Note that it doesn't improve anything.

# Expected behavior
Improvement.

# Screenshots
![screenshot-shit](https://user-images.githubusercontent.com/31852/117393899-9fb2be00-aee4-11eb-84ea-a8e2bd87996f.png)

# Versions and configuration
- OS: Linux
    - OS Version: <unknown>
- Game Version: 0.E-10491-g7c20c92f7e-dirty [64-bit]
- Graphics Version: Curses
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions],
    Alternative Map Key [alt_map_key],
    No Rail Stations [No_Rail_Stations]
]

# Additional context

[2021.04.25.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/6438828/2021.04.25.zip)

**Comment by actual-nh** [comment]
> Can confirm on `0.E-10474-g1ae66c97a0 [64-bit]` but I don't think this is really a bug.
> It has more to deal with how armor values are calculated, the material thickness of `0.3` is probably too small for the clothing mods to have an effect on, ( increasing it to around ~1 is enough to see an effect ).

In other words, this is a roundoff problem regarding display of the changes?

**Comment by Saicchi** [comment]
I haven't done much testing, the values are a float so there shouldn't be rounding errors, but when changing the values from 0.1 ~ 1.0, on both the rain hood and a leather gloves, I couldn't get any extra protection unless the material thickness was above 1.
tl;dr: modifying clothing with `material_thickness` below 1 doesn't seem to be giving any protection.

**Comment by Anaglyphic** [comment]
Hmm. As discussions go for feature requests that are entered as issues... this might be one of those _sounds simple enough_ things, but probably requires entire clothing / armour system re-factoring.

Like the OP says, try to reinforce say a knit hat, or cargo pants, or leather bikini even, with fur, leather, kevlar, whatever... nope No dice. No change in stats on the Modify Clothing pop-up, although it will apparently allow you to? I guess if you want to waste resources.

But tag your headlamp, binoculars, or your black leather trench, Blade... all of those are possible targets, with increased stats and encumbrance to go with. The difference is one single word in the item JSON array: **armor**. If an object has "type": "ARMOR", or "type": "TOOL_ARMOR", it's reinforcable.

But - arbitrarily changing these things, what _should be_ armor and _should not be_ armor to be just clothes and/or reinforcable, may have rippling repercussions on gameplay, especially early on. I would not recommend suggesting changing this until you get a few veteran cdda brains to discuss the implication(s).

( Nothing is stopping _you_ from editing _your_ local data/, of course. Ceramic ballistic plates in your boxers? Go for it! )

**Comment by actual-nh** [comment]
The bug would appear to be that the interface doesn't tell you the problem. (There may be another issue to this effect, but I'm not finding it.)

**Comment by Night-Pryanik** [comment]
No longer applicable after #58035.
